pkg/operator/ceph/object/objectstore.go

```diff
@@ -757,15 +759,44 @@ func enableRGWDashboard(context *Context) error {
 	if err != nil || errCode != 0 {
 		return errors.Wrapf(err, "failed to create user %q", DashboardUser)
 	}
-	args := []string{"dashboard", "set-rgw-api-access-key", *u.AccessKey}
-	cephCmd := ceph.NewCephCommand(context.Context, context.clusterInfo, args)
+
+	filePasswordSupported := mgr.FileBasedPasswordSupported(context.clusterInfo)
+	var accessArgs, secretArgs []string
+	var secretFile *os.File
+
+	// for latest Ceph versions
+	if filePasswordSupported {
+		accessFile, err := mgr.CreateTempPasswordFile(*u.AccessKey)
+		if err != nil {
+			return errors.Wrap(err, "failed to create a temporary dashboard access-key file")
+		}
+
+		accessArgs = []string{"dashboard", "set-rgw-api-access-key", "-i", accessFile.Name()}
+		defer func() {
+			if err := os.Remove(accessFile.Name()); err != nil {
+				logger.Errorf("failed to clean up dashboard access-key file. %v", err)
+			}
+		}()
+
+		secretFile, err = mgr.CreateTempPasswordFile(*u.SecretKey)
```

Comment:
Is this secretFile being removed? Looks like that code was removed.

Reply:
Yes, because in case of else...it will try to remove the file of *os.file type even if it is not present...so I removed it.

Reply:
Seems like we still need to remove the file. What if we remove the file in the goroutine, but first just check for `if filePasswordSupported`?